src/main/java/land/oras/credentials/FileStore.java

```diff
@@ -0,0 +1,136 @@
+
+    public static final String ERR_PLAINTEXT_PUT_DISABLED = "Putting plaintext credentials is disabled";
+    public static final String ERR_BAD_CREDENTIAL_FORMAT = "Bad credential format";
+
+    /**
+     * Constructor for FileStore.
+     *
+     * @param disablePut boolean flag to disable putting credentials in plaintext.
+     * @param config configuration instance.
+     */
+    public FileStore(boolean disablePut, Config config) {
+        this.disablePut = disablePut;
+        this.config = Objects.requireNonNull(config, "Config cannot be null");
+    }
+
+    /**
+     * Creates a new FileStore based on the given configuration file path.
+     *
+     * @param configPath Path to the configuration file.
+     * @return FileStore instance.
+     * @throws Exception if loading the configuration fails.
+     */
+    public static FileStore newFileStore(String configPath) throws Exception {
+        Config cfg = Config.load(configPath);
+        return new FileStore(false, cfg);
+    }
+
+    /**
+     * Retrieves credentials for the given server address.
+     *
+     * @param serverAddress Server address.
+     * @return Credential object.
+     * @throws Exception if retrieval fails.
+     */
+    public Credential get(String serverAddress) throws Exception {
+        return config.getCredential(serverAddress);
+    }
+
+    /**
+     * Saves credentials for the given server address.
+     *
+     * @param serverAddress Server address.
+     * @param credential Credential object.
+     * @throws Exception if saving fails.
+     */
+    public void put(String serverAddress, Credential credential) throws Exception {
+        if (disablePut) {
+            throw new UnsupportedOperationException(ERR_PLAINTEXT_PUT_DISABLED);
+        }
+        validateCredentialFormat(credential);
+        config.putCredential(serverAddress, credential);
+    }
+
+    /**
+     * Deletes credentials for the given server address.
+     *
+     * @param serverAddress Server address.
+     * @throws Exception if deletion fails.
+     */
+    public void delete(String serverAddress) throws Exception {
+        config.deleteCredential(serverAddress);
+    }
+
+    /**
+     * Validates the format of the credential.
+     *
+     * @param credential Credential object.
+     * @throws Exception if the credential format is invalid.
+     */
+    private void validateCredentialFormat(Credential credential) throws Exception {
+        if (credential.getUsername().contains(":")) {
+            throw new IllegalArgumentException(ERR_BAD_CREDENTIAL_FORMAT + ": colons(:) are not allowed in username");
+        }
+    }
+
+    /**
+     * Nested Config class for configuration management.
+     */
+    public static class Config {
+        private final ConcurrentHashMap<String, Credential> credentialStore = new ConcurrentHashMap<>();
+
+        public static Config load(String configPath) throws Exception {
+            // Simulate loading the configuration file.
```

Comment:
So we don't read anything?

Reply:
Added implementation of this method thanks : )